geomstats/geometry/siegel.py

```diff
@@ -0,0 +1,614 @@
+            Tangent vector at zero.
+        base_point : array-like, shape=[..., n, n]
+            Point on the Siegel space.
+
+        Returns
+        -------
+        tangent_vec_at_zero : array-like, shape=[..., n, n]
+            Tangent vector at zero (null matrix).
+        """
+        data_type = (tangent_vec + base_point).dtype
+        identity = create_identity_mat(base_point.shape, dtype=data_type)
+        base_point_transconj = ComplexMatrices.transconjugate(base_point)
+        aux_1 = gs.einsum("...ij,...jk->...ik", base_point, base_point_transconj)
+        aux_2 = gs.einsum("...ij,...jk->...ik", base_point_transconj, base_point)
+        aux_3 = identity - aux_1
+        aux_4 = identity - aux_2
+        factor_1 = HermitianMatrices.powerm(aux_3, -1 / 2)
+        factor_3 = HermitianMatrices.powerm(aux_4, -1 / 2)
+        prod_1 = gs.einsum("...ij,...jk->...ik", factor_1, tangent_vec)
+        tangent_vec_at_zero = gs.einsum("...ij,...jk->...ik", prod_1, factor_3)
+        return tangent_vec_at_zero
+
+    @staticmethod
+    def exp_at_zero(tangent_vec):
+        """Compute the Siegel exponential map at zero.
+
+        Compute the exponential map at zero (null matrix) of tangent_vec.
+
+        Parameters
+        ----------
+        tangent_vec : array-like, shape=[..., n, n]
+            Tangent vector at base point.
+
+        Returns
+        -------
+        exp : array-like, shape=[..., n, n]
+            Point on the manifold.
+        """
+        data_type = tangent_vec.dtype
+        identity = create_identity_mat(tangent_vec.shape, dtype=data_type)
+        tangent_vec_transconj = ComplexMatrices.transconjugate(tangent_vec)
+        aux_1 = gs.einsum("...ij,...jk->...ik", tangent_vec, tangent_vec_transconj)
+        aux_2 = HermitianMatrices.powerm(aux_1, 1 / 2)
+        aux_3 = HermitianMatrices.expm(2 * aux_2)
+        factor_1 = aux_3 - identity
+        aux_4 = aux_3 + identity
+        factor_2 = HermitianMatrices.powerm(aux_4, -1)
+        factor_3 = HermitianMatrices.powerm(aux_2, -1)
+        prod_1 = gs.einsum("...ij,...jk->...ik", factor_1, factor_2)
+        prod_2 = gs.einsum("...ij,...jk->...ik", prod_1, factor_3)
+        exp = gs.einsum("...ij,...jk->...ik", prod_2, tangent_vec)
+        return exp
+
+    @staticmethod
+    def isometry(point, point_to_zero):
+        """Define an isometry for the Siegel metric.
+
+        Isometry for the Siegel metric sending point_to_zero
+        (parameter of the isometry) on zero.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., n, n]
+            Point on the Siegel space.
+        point_to_zero : array-like, shape=[..., n, n]
+            Point send on zero (null matrix) by the isometry.
+
+        Returns
+        -------
+        point_image : array-like, shape=[..., n, n]
+            Image of point by the isometry.
+        """
+        data_type = (point + point_to_zero).dtype
```

Comment:
use `point`. no need to create the variable.

Reply:
Done.